packages/core/src/Cardano/types/Utxo.ts

```diff
@@ -15,7 +16,30 @@ export interface HydratedTxIn extends TxIn {
 export interface TxOut {
   address: Address;
   value: Value;
-  datum?: Hash32ByteBase16;
+
+  /**
+   * Datum hash, this allows to specify a Datum without publicly revealing its value. To spend an output which specifies
+   * this type of datum, the actual Datum value must be provided and will be added to the witness set of
+   * the transaction.
+   */
+  datumHash?: Hash32ByteBase16;
+
+  /**
+   * Inline datum reveals the value on the blockchain at the time of output creation. This way of attaching datums lets
+   * users consume this output without specifying the datum value.
+   */
+  inlineDatum?: HexBlob;
+
+  /**
+   * Reference scripts can be used to satisfy script requirements during validation, rather than requiring the spending
+   * transaction to do so. This will allow transactions using common scripts to be much smaller.
+   *
+   * The key idea is to use reference inputs and outputs which carry actual scripts ("reference scripts"), and allow
+   * such reference scripts to satisfy the script witnessing requirement for a transaction. This means that the
+   * transaction which uses the script will not need to provide it at all, so long as it referenced an output
+   * which contained the script.
+   */
+  scriptRef?: Script;
```

Comment:
This should probably be a different type
```suggestion
  scriptRef?: ScriptReference;
```

Reply:
Oh so it's actually the 'Script' type. Maybe let's go back to using the existing `Script` type then, but also rename the field? Like this:
```ts
script: Script
```
Then, how is it actually referenced (used as witness)?

Reply:
> but if we want to change it to just script I am also okay with that

I can't think of a reason to have a separate type if it refers to the same thing, so yeah, let's do that unless there's a reason to have separate types.

> ...as a reference input and the node will search for the eUTXO and fetch the script automatically. What I have done here allows us to create these reference outputs (or inputs), but we are still missing the part of using them, that will come once we have smart-contract support

Thanks for an explanation. The part of using them is what's unclear to me, are all scripts in reference inputs implicitly used as witness? :thinking: